src/IO/RemoteReadBufferCache.cpp

```diff
@@ -172,35 +169,44 @@ void RemoteCacheController::backgroundDownload()
     }
     std::unique_lock lock(mutex);
     current_offset += total_bytes;
-    file_meta_data.status = RemoteFileMetadata::DOWNLOADED;
+    file_status = DOWNLOADED;
     flush(true);
     data_file_writer.reset();
-    remote_read_buffer.reset();
     lock.unlock();
     more_data_signal.notify_all();
-    RemoteReadBufferCache::instance().updateTotalSize(file_meta_data.file_size);
-    LOG_TRACE(log, "Finish download into local path: {}, file meta data:{} ", local_path.string(), file_meta_data.toString());
+    RemoteReadBufferCache::instance().updateTotalSize(file_meta_data_ptr->getFileSize());
+    LOG_TRACE(log, "Finish download into local path: {}, file meta data:{} ", local_path.string(), file_meta_data_ptr->toString());
 }
 
-void RemoteCacheController::flush(bool need_flush_meta_data_)
+void RemoteCacheController::flush(bool need_flush_status)
 {
     if (data_file_writer)
     {
         data_file_writer->sync();
     }
-
-    if (!need_flush_meta_data_)
-        return;
-
-    file_meta_data.save(local_path / "meta.txt");
+    if (need_flush_status)
+    {
+        Poco::JSON::Object jobj;
+        jobj.set("file_status", static_cast<Int32>(file_status));
+        jobj.set("meta_data_class", meta_data_class);
```

Comment:
I don't like this idea of writing metadata class name into metadata. Why can't we just start loading storage specific metadata at storage startup? Then we will know the needed type.

Reply:
How to you know which class it is

Reply:
StorageHive -> StorageHiveMetadata.
StorageHDFS -> StorageHDFSMetadata
StorageS3 -> StorageS3Metadata

Reply:
So I don't realy know your point here.

The RemoteReadBufferCache is a global instance, and not  bound to any specified storage.  It works on the ReadBuffer level.
In general for managing the local cache copacity, we just need to know
1. the file size 
2.  whether it is changed after the last download. for dropping out the old version

Reply:
My point is to avoid hive specific `schema` and `cluster` mentions in `RemoteReadBufferCache`.

Reply:
done